src/ai/backend/common/types.py

```diff
@@ -1581,6 +1581,21 @@ def profile_target(self, role: RedisRole) -> RedisTarget:
             return self._override_targets[role]
         return self._base_target
 
+    @staticmethod
+    def _parse_addr(addr_data) -> HostPortPair:
+        addr = None
+        if isinstance(addr_data, HostPortPair):
+            addr = HostPortPair(addr_data.host, addr_data.port)
+        elif isinstance(addr_data, Mapping):
+            addr = HostPortPair(addr_data["host"], addr_data["port"])
+        elif isinstance(addr_data, tuple):
+            addr = HostPortPair(addr_data[0], addr_data[1])
+        else:
+            addr_data_parts = addr_data.split(":")
+            addr = HostPortPair(addr_data_parts[0], int(addr_data_parts[1]))
```

Comment:
Let's use `match ... case` syntax when possible to make our mental models clearer.